src/config/loading.rs

```diff
@@ -97,8 +97,16 @@ pub fn process_paths(config_paths: &[ConfigPath]) -> Option<Vec<ConfigPath>> {
     Some(paths)
 }
 
-pub fn load_from_paths(config_paths: &[ConfigPath]) -> Result<Config, Vec<String>> {
-    let pipelines = load_pipelines_from_paths(config_paths)?;
+pub fn load_from_paths(
+    config_paths: &[ConfigPath],
+    pipeline_paths: &[PathBuf],
+) -> Result<Config, Vec<String>> {
+    let pipelines = if pipeline_paths.is_empty() {
+        let pipeline_paths = pipeline_paths_from_config_paths(config_paths);
+        load_pipelines_from_paths(&pipeline_paths)?
+    } else {
+        load_pipelines_from_paths(pipeline_paths)?
+    };
```

Comment:
Would it be worth extracting this pattern to its own function? Seems like we're doing this a lot. Maybe something like `load_pipelines_from_paths(pipeline_paths, config_paths)`?